external/kafka-0-10-sql/src/main/scala/org/apache/spark/sql/kafka010/KafkaOffsetReader.scala

```diff
@@ -110,6 +110,34 @@ private[kafka010] class KafkaOffsetReader(
     kafkaReaderThread.shutdown()
   }
 
+  /**
+   * Commits offsets to the consumer_offsets topic
+   * for the driver consumer.
+  */
+  def commitOffsetsAsync(
+      partitionToOffsets: Map[TopicPartition, Long],
+      s: KafkaCommitsSource)
+    : Unit = {
+    if (_consumer != null) runUninterruptibly {
+     val commitCallback = new OffsetCommitCallback() {
```

Comment:
Absolutely not sure just asking. Do we need new instance all the time?

Reply:
Probably not will check.